engine/classes/Elgg/EntityIconService.php

```diff
@@ -435,6 +435,11 @@ public function getSizes($entity_type = null, $entity_subtype = null, $type = 'i
 			$sizes = $this->hooks->trigger("entity:$type:sizes", $entity_type, $params, $sizes);
 		}
 
+		if (empty($sizes)) {
```

Comment:
Logging an error for an empty array is fine. I am thinking throwing if hook returns something other than an array.